ethexe/db/src/mem.rs

```diff
@@ -32,13 +32,13 @@ impl CASDatabase for MemDb {
         Box::new(self.clone())
     }
 
-    fn read(&self, hash: &H256) -> Option<Vec<u8>> {
+    fn read(&self, hash: H256) -> Option<Vec<u8>> {
         let key = hash.as_bytes().to_vec();
         self.inner.get(&key).map(|v| v.value().clone())
     }
 
-    fn write_by_hash(&self, hash: &H256, data: &[u8]) {
-        debug_assert_eq!(*hash, crate::hash(data));
+    fn write_by_hash(&self, hash: H256, data: &[u8]) {
+        debug_assert_eq!(hash, crate::hash(data));
```

Comment:
btw I'd keep this assertion even in prod. it's critical for security to maintain db hashes invariants

Reply:
Keeping this in produciton would make this method useless. Because the only one difference it can provide for us is: whether we need to calculate hash or not. Actually we do not uses it anyway))